pkg/build/definition_parser_deffile.go

```diff
@@ -9,13 +9,156 @@ package build
 
 import (
 	"bufio"
+	"bytes"
+	"fmt"
 	"io"
 	"log"
 	"regexp"
 	"strings"
 	"unicode"
 )
 
+// validSections just contains a list of all the valid sections a definition file
+// could contain. If any others are found, an error will generate
+var validSections = map[string]bool{
+	"help":        true,
+	"setup":       true,
+	"files":       true,
+	"labels":      true,
+	"environment": true,
+	"pre":         true,
+	"post":        true,
+	"runscript":   true,
+	"test":        true,
+}
+
+// scanSections is the SplitFunc for the scanner that will parse the deffile. It will split into tokens
+// that designated by a line starting with %
+// If there are any Golang devs reading this, please improve your documentation for this. It's awful.
+func scanSections(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
You may only specify the output types, without the names.
```go
func scanSections(data []byte, atEOF bool) (int,  []byte,  error)
```

Reply:
Also I don't understand how this func is an enhanced work of the ones that were already for parsing the deffile.

Reply:
@ArangoGutierrez  - you can name outputs... and you then get naked return, and built-in doc:

https://golang.org/doc/effective_go.html#named-results

Reply:
I know you can, but is cleaner, that way. you may define the name after, when calling the func

Reply:
I really think having the name in the function signature for functions with compound returns helps a lot... it's major win when you have functions returning multiple things of same type e.g. (string, string, string)... which one is which thing?

Reply:
I don't get to decide the signature of `splitSection`, it is necessary to be of type [SplitFunc](https://golang.org/pkg/bufio/#SplitFunc), so I am just using the signature from there.

In general, I don't mind naming function return values. You can still return a variable with a different name that shares the same type, and you still define the name when calling the function and assigned the return values. However, it does two things:

1. Clarifies the intended usage for the return values
2. Allows you to do set the values of the named return values and simply call `return` at the end to return the values of those variables. 

I don't think we should shy away from naming return values.